Added URLGenerate and URLQueryFromJSON builtins for URL generation

vrnmthr opened the pull request:
Wrote two new builtins:

urlquery.fromjson takes as input a JSON map of key value pairs, where the values can either be single strings or array of strings, and produces a properly encoded query string (minus the ?). 

url.generate takes as input a host, a path and a JSON object (empty object for no query string) and produces a properly formatted http:// URL 

Signed-off-by: vrnmthr <varun.mathur@live.com>

tsandall commented on ast/builtins.go:
```diff
@@ -720,6 +722,28 @@ var URLQueryEncode = &Builtin{
+}
+
+// URLGenerate takes a host and a path and a list of query arguments in JSON format and encodes it into a string
+var URLGenerate = &Builtin{
```
I'm hesitant to add this built-in function for two reasons.

1. It assumes that users don't need to control the scheme or the port. We could extend it to support those, but what about other cases? E.g., sometimes you may not want the URL parameters encoded.

2. It's possible to implement both fo these new built-in functions in Rego.

vrnmthr commented:
I removed the URLGenerate function as requested and modified the name for the URLQuery one. I agree that it is possible to write the URLQuery one in rego, but it is nontrivial and I think it would benefit from the builtin. I agree with scrapping URLgenerate though :)
